### PR TITLE
fix(scoop-bucket): Check for git before running command

### PR DESCRIPTION
**Pull request by rashil2000**
Hotfix for #4704

### CODE REVIEW

File: libexec/scoop-bucket.ps1 [hunk 1]
```diff
@@ -38,11 +38,11 @@ function list_buckets {
             Measure-Object | Select-Object -ExpandProperty Count
         )
         $updated = 'N/A'
-        if (Test-Path (Join-Path $source '.git')) {
-            $updated = git -C "`"$source`"" log --date=format:"`"%Y-%m-%d %H:%M:%S`"" --format='%ad' -n 1
+        if ((Test-Path (Join-Path $source '.git')) -and (Get-Command git -ErrorAction Ignore)) {
```
Comment (niheaven) [inline libexec/scoop-bucket.ps1:41]:
```suggestion
        if ((Test-Path (Join-Path $source '.git')) -and (Get-Command git -ErrorAction SilentlyContinue)) {
```

`$Error` could be used. `SilentlyContinue` is always preferred than `Ignore`.

**Comment by niheaven** [comment]
The code LGTM, please update CHANGLOG

**Comment by HUMORCE** [comment]
```
> scoop bucket list
Get-Date: C:\Users\HUMOR\scoop\apps\scoop\current\libexec\scoop-bucket.ps1:42
Line |
  42 |  …  --date=format:"`"%d-%m-%Y %H:%M:%S`"" --format='%ad' -n 1 | Get-Date
     |                                                                 ~~~~~~~~
     | The input object cannot be bound to any parameters for the command either because the command does not
     | take pipeline input or the input and its properties do not match any of the parameters that take
     | pipeline input.
...
```

**Comment by rashil2000** [comment]
image

image


How to reproduce?

**Comment by HUMORCE** [comment]
screenshot_20220224164358

**Comment by HUMORCE** [comment]
> %ad
> author date (format respects --date= option)
> 
> %aD
> author date, RFC2822 style

if date formatting is not required, use `%aD`:

```
> git log --format='%aD' -n 1 | Get-Date

Thursday, February 24, 2022 12:03:00 PM
```

**Comment by rashil2000** [comment]
image

image

PowerShell 5.1

**Comment by rashil2000** [comment]
> if date formatting is not required, use `%aD`:

Hmm, okay. Seems fine by me. Can you fix it?

**Comment by HUMORCE** [comment]
screenshot_20220224170745

related to OS's datetime format. (unconfirmed)

**Pull request merged**